src/mono/sample/wasm/browser2/main.js

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import DotNet from "./dotnet.js"; // User can import DotNet so that there can even potentially be multiple instances in 1 app
+
+// Called when the mono-config.js (soon to be mono-config.json via PR#53606) is loaded
+function onConfigLoaded (config) {
+    // we can do some modification to config at runtime here
+    // then return it
+    // i.e. config.newProp = "test new prop";
+
+    return config;
+}
+
+// Called when Emscripten runtime is loaded
+function onRuntimeInitialized () {
+}
+
+// Called when MONO runtime is loaded and ready or if it errors
+function onMonoRuntimeInitialized (error) {
```

Comment:
Let's split this. Remove the error parameter and call it only on success.
Let's have another callback `onFatalError` which could be called any time runtime experienced something terrible.

Reply:
This would result in 2 required callbacks. Perhaps, we should define some priorities for these changes? Currently I was just trying to make it as elegant as possible but elegant is a subjective term and not overly precise haha. So I was trying to minimize API surface area and make things explicit. Are these the only goals or are there some more as well?

Regardless, I agree with making a `onFatalError` callback.

Reply:
onFatalError should be optional too. default behavior `console.error` if not defined